src/java.base/share/classes/java/util/zip/CRC32C.java

```diff
@@ -222,9 +222,9 @@ private static int updateBytes(int crc, byte[] b, int off, int end) {
         if (end - off >= 8 && Unsafe.ARRAY_BYTE_INDEX_SCALE == 1) {
 
             // align on 8 bytes
-            int alignLength
+            long alignLength
                     = (8 - ((Unsafe.ARRAY_BYTE_BASE_OFFSET + off) & 0x7)) & 0x7;
-            for (int alignEnd = off + alignLength; off < alignEnd; off++) {
+            for (long alignEnd = off + alignLength; off < alignEnd; off++) {
```

Comment:
I think casting the (now) long expression back to `int` makes more sense here.

Reply:
alignLength is a very small value, which is an int, but alignEnd needs to be a long. Here, changing alignLength to a long can avoid alignEnd overflow.